### PR TITLE
Feature/compliance

### PR DESCRIPTION
**Pull request by EvgenKor**
New CLI commands:
- `compliance-report` to generate report within specified time interval. Report displayed in the console and can be downloaded
- `account-id` returns unique ID of your account (sk hash)

**Here is report fragment:**
image

**Here is full report example:**
[c5a88b46-0ab4-4630-a201-5eed59d7316e.zip](https://github.com/zkBob/zkbob-console/files/10523600/c5a88b46-0ab4-4630-a201-5eed59d7316e.zip)

**Single file report format:**
[26939d7e-e30f-4e3c-87a4-7eefa028239d.zip](https://github.com/zkBob/zkbob-console/files/10527693/26939d7e-e30f-4e3c-87a4-7eefa028239d.zip)

### CODE REVIEW

File: src/commands.ts [hunk 1]
```diff
@@ -760,6 +762,175 @@ function humanReadable(record: HistoryRecord, denominator: number): string {
...
+    this.echo(`Report was generated at: [[;white;]${reportGeneratedDate.toLocaleString()}] (${reportGeneratedDate.getTime()})`);
+
+    this.echo(`You could also [[!;;;;${reportUrl}]download raw report] now`); 
+
+    this.resume();
+    
+}
+
+function findChunk(index: number, array: Array<{data: Uint8Array, index: number}>): Uint8Array | undefined {
+    for(const obj of array) {
```
Comment (k1rill-fedoseev) [inline src/commands.ts]: minor: `array.find(obj => obj.index == index)?.data` should do the same

File: src/commands.ts [hunk 2]
```diff
@@ -760,6 +762,175 @@ function humanReadable(record: HistoryRecord, denominator: number): string {
...
+        } else if (value instanceof Object.getPrototypeOf(Uint8Array)) {
+            return `${bufToHex(value)}`;
+        }  else {
+            return value;
+        }
+    }
+
+    let zipFile = new JSZip();
+    zipFile.file(`metadata.json`, JSON.stringify(metadata, replacer, space));
+    zipFile.file(`transactions.json`, JSON.stringify(report, replacer, space));
```
Comment (k1rill-fedoseev) [inline src/commands.ts]: Why do we need 2 separate files?

**Review by k1rill-fedoseev — APPROVED** [review]

File: src/commands.ts (hunk 1 above)
Comment (EvgenKor) [reply src/commands.ts]: Yes, strongly agreed :) Fixed in https://github.com/zkBob/zkbob-console/pull/57/commits/e29de03e3d1f965765e97594c5a5ec4a5cde866b

File: src/commands.ts (hunk 2 above)
Comment (EvgenKor) [reply src/commands.ts]: Actually it's just an export example. The compliance report design will be formed by application. So I think it doesn't matter. Currently I rewrote it to export a single file at https://github.com/zkBob/zkbob-console/commit/e29de03e3d1f965765e97594c5a5ec4a5cde866b
A little benefit of two files: macOS unpack every report in the separate folders :)

New report format: 
[26939d7e-e30f-4e3c-87a4-7eefa028239d.zip](https://github.com/zkBob/zkbob-console/files/10527685/26939d7e-e30f-4e3c-87a4-7eefa028239d.zip)

**Review by AllFi — APPROVED** [review]
Looks good to me

**Comment by EvgenKor** [comment]
Adapted for the DD compliance records. Due to DD records are unencrypted, we export just input notes. Looks like following: 
image

**Review by AllFi — APPROVED** [review]
It works properly and looks good to me.

**Review by k1rill-fedoseev — APPROVED** [review]

**Review by AllFi — APPROVED** [review]

**Pull request merged**